### PR TITLE
feat(unlock-app) - customize name of sender

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

Add `email sender` to email options in settings page

**Email options** 

Screenshot 2023-05-02 at 15 34 30

**Email options hidden when email are disabled** 

Screenshot 2023-05-02 at 16 01 56

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/unlock-js/src/abis.ts [hunk 1]
```diff
@@ -1043,7 +1043,7 @@ const abis = {
         "function tokenOfOwnerByIndex(address _keyOwner,uint256 _index) view returns (uint256)",
```
Comment (kalidiagne) [inline packages/unlock-js/src/abis.ts:1043]: @julien51 we need this one? update after build of `unlock-js`

File: unlock-app/src/components/interface/locks/Settings/forms/EmailSettingsForm.tsx [hunk 2]
```diff
@@ -57,28 +60,41 @@ export const EmailReplyToForm = ({
...
-              placeholder="your@email.com"
-              label="Reply-to"
-              disabled={disabledInput || !enabled}
-              {...register('replyTo')}
+              type="text"
+              placeholder="Example Name"
+              label="Email sender"
+              disabled={disabledInput}
+              autoComplete="off"
+              description="Set the email sender that will appear on the email sent."
```
Comment (kalidiagne) [inline unlock-app/src/components/interface/locks/Settings/forms/EmailSettingsForm.tsx:95]: @julien51 can we improve copy here?

File: locksmith/src/controllers/v2/lockSettingController.ts [hunk 3]
```diff
@@ -21,6 +21,11 @@ const LockSettingSchema = z.object({
       description: 'Credit card default price to use on checkout.',
     })
     .optional(),
+  emailSender: z
+    .string({
+      description: 'Custom name for email sender',
```
Comment (kalidiagne) [inline locksmith/src/controllers/v2/lockSettingController.ts:26]: @julien51 can we improve copy here?

Comment (julien51) [inline locksmith/src/controllers/v2/lockSettingController.ts:26]:
```suggestion
      description: 'Custom name used as the email sender. This is the name that most email clients will show to readers.',
```

File: packages/unlock-js/src/abis.ts (hunk 1 above)
Comment (julien51) [reply packages/unlock-js/src/abis.ts:1043]: Hum, let's remove from this PR please.

File: unlock-app/src/components/interface/locks/Settings/forms/EmailSettingsForm.tsx (hunk 2 above)
Comment (julien51) [reply unlock-app/src/components/interface/locks/Settings/forms/EmailSettingsForm.tsx:95]: I think that's good!

File: wedlocks/src/route.js [hunk 4]
```diff
@@ -61,6 +61,7 @@ export const route = async (args) => {
     from: config.sender,
     to: args.recipient,
     replyTo: args?.replyTo || undefined,
+    sender: args?.emailSender || undefined,
```
Comment (julien51) [inline wedlocks/src/route.js:64]:
```suggestion
    sender: args?.emailSender ||"Unlock Labs",
```

**Review by julien51 — APPROVED** [review]
Please do test on staging thoroughly!

File: wedlocks/src/route.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (julien51) [reply wedlocks/src/route.js:64]: @kalidiagne Are you sure this `sender` param exists???

https://nodemailer.com/message/

[the remark this answers is not in the document]
Comment (julien51) [reply wedlocks/src/route.js:64]: When I read the docs I think this is wrong!

```
sender - An email address that will appear on the Sender: field (always prefer from if you’re not sure which one to use)
```

[the remark this answers is not in the document]
Comment (kalidiagne) [reply wedlocks/src/route.js:64]: yes! 

Screenshot 2023-05-02 at 16 02 05

[the remark this answers is not in the document]
Comment (julien51) [reply wedlocks/src/route.js:64]: You need to format the `from` param:

```
from - The email address of the sender. All email addresses can be plain ‘[sender@server.com](mailto:sender@server.com)’ or formatted '“Sender Name” [sender@server.com](mailto:sender@server.com)', see [Address object](https://nodemailer.com/message/addresses/) for details
```

**Review by julien51 — CHANGES_REQUESTED** [review]
I am pretty sure `sender` is wrong after reviewing the nodemailer docs

File: wedlocks/src/route.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (kalidiagne) [reply wedlocks/src/route.js]: Updated!

File: packages/unlock-js/src/abis.ts [hunk 5]
```diff
@@ -1043,7 +1043,7 @@ const abis = {
...
         "function tokenURI(uint256 _tokenId) view returns (string)",
         "function totalKeys(address _keyOwner) view returns (uint256)",
-        "function totalSupply() view returns (uint256 _totalKeysCreated)",
+        "function totalSupply() view returns (uint256)",
```
Comment (kalidiagne) [inline packages/unlock-js/src/abis.ts:1046]:
```suggestion
        "function totalSupply() view returns (uint256 _totalKeysCreated)",
```

File: packages/unlock-js/src/abis.ts [hunk 6]
```diff
@@ -1055,7 +1055,7 @@ const abis = {
         "function updateTransferFee(uint256 _transferFeeBasisPoints)",
         "function withdraw(address _tokenAddress,address _recipient,uint256 _amount)"
       ],
-      "bytecodeHash": "0x80041134ce431f1adb2b2dfe00df82753d269761cbf9bac1232e0523fa787ed2"
+      "bytecodeHash": "0x363551a7e8051fe1a13a8430e2277433660616ebf20394bf1c7c7001f949e2d7"
```
Comment (kalidiagne) [inline packages/unlock-js/src/abis.ts:1058]:
```suggestion
      "bytecodeHash": "0x80041134ce431f1adb2b2dfe00df82753d269761cbf9bac1232e0523fa787ed2"
```

**Pull request merged**